app/src/org/commcare/views/widgets/MediaWidget.java

```diff
@@ -210,7 +210,7 @@ public void setBinaryData(Object binaryuri) {
             newMedia = new File(destMediaPath);
         } else {
             recordedFileName = FileUtil.getFileName(binaryPath);
-            destMediaPath = mInstanceFolder + System.currentTimeMillis() + customFileTag + FileUtil.getExtension(binaryPath);
+            destMediaPath = mInstanceFolder + System.currentTimeMillis() + customFileTag + "." + FileUtil.getExtension(binaryPath);
```

Comment:
@wpride FileUtil can return a blank extension if the path doesn't have one, which could result in files like
```
something.
```

We should make sure we only add the dot if the extension is getting added to a non-blank value.

Reply:
Thanks Clayton, nice catch. Seems like we're a bit fast and loose with the path and extension handling and much of this code could be centralize.

My fix implements the same assertion as [here](https://github.com/dimagi/commcare-android/blob/d270afa3b30b10a92c0c73a9bffecad48aeeb54c/app/src/org/commcare/activities/FormEntryActivity.java#L356-L356) - ensuring that the media type is one that we can actually upload to the server